src/Support/OperationExtensions/ResponderExtension.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Dedoc\Scramble\Support\OperationExtensions;
+
+use Dedoc\Scramble\Extensions\OperationExtension;
+use Dedoc\Scramble\Support\Generator\Combined\AnyOf;
+use Dedoc\Scramble\Support\Generator\Operation;
+use Dedoc\Scramble\Support\Generator\Response;
+use Dedoc\Scramble\Support\Generator\Schema;
+use Dedoc\Scramble\Support\Generator\Types as OpenApiTypes;
+use Dedoc\Scramble\Support\RouteInfo;
+use Dedoc\Scramble\Support\Type\Generic;
+use Illuminate\Support\Collection;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Return_;
+
+class ResponderExtension extends OperationExtension
+{
+    public function handle(Operation $operation, RouteInfo $routeInfo)
+    {
+        $definition = $routeInfo->methodNode();
+
+        if (!$definition instanceof ClassMethod) {
+            return;
+        }
+
+        $statements = $definition->getStmts();
+
+        $returnStatements = array_filter($statements, function ($statement) {
+            return $statement instanceof Return_
+                && $statement->expr instanceof \PhpParser\Node\Expr\MethodCall
+                && $this->usesResponderClass($statement->expr);
+        });
+
+        $responses = collect($returnStatements)
+            ->map(function ($returnStatement) {
```

Comment:
Do you know the type of `$returnStatement`?

Reply:
Nope

Reply:
```suggestion
            ->map(function (mixed $returnStatement) {
```